### PR TITLE
TP2000-432 QuotaAssociation objects have been created with the same main and sub quota

### PR DESCRIPTION
**Pull request by gabelton**
# TP2000-432 QuotaAssociation objects have been created with the same main and sub quota

## Why

It shouldn't be possible to create an association where the same definition is both the parent and the child.

It shouldn't be possible to create overlapping definitions for the same quota order number.

Therefore, we should formalise the above as business rules.

## What

Adds two new rules, tests, and updates model business rules lists

### CODE REVIEW

File: quotas/business_rules.py [hunk 1]
```diff
@@ -250,6 +250,24 @@ def get_relation_model(self, quota_definition):
         return quota_definition.quotablocking_set.model
 
 
+class OverlappingQuotaDefinition(BusinessRule):
```
Comment (paulpepper-trade) [inline quotas/business_rules.py:253]: Do we distinguish between TARIC business rules and Tamato's own, non-TARIC biz rules in any way? If not, is there value in doing so?

File: quotas/business_rules.py [hunk 2]
```diff
@@ -250,6 +250,24 @@ def get_relation_model(self, quota_definition):
...
+    """There may be no overlap in time of two quota definitions with the same
+    quota order number id."""
+
+    def validate(self, quota_definition):
```
Comment (paulpepper-trade) [inline quotas/business_rules.py:257]: Perhaps worth adding some type hinting to the params of these `validate` methods.

**Review by paulpepper-trade — APPROVED** [review]

File: quotas/business_rules.py (hunk 1 above)
Comment (gabelton) [reply quotas/business_rules.py:253]: The only distinction I'm aware of is that TARIC rules follow the alphanumeric "ME32" format and our own rules take the form of a description of the violation. There may be exceptions to that, but I think that's the general principle

File: quotas/business_rules.py (hunk 2 above)
Comment (gabelton) [reply quotas/business_rules.py:257]: I think the issue is that we get a circular import error when specifying QuotaDefinition, which I suppose is why the other rules in this file don't use it

Comment (paulpepper-trade) [reply quotas/business_rules.py:257]: Could we use forward references?
https://peps.python.org/pep-0484/#forward-references

File: quotas/business_rules.py (hunk 1 above)
Comment (SteveCN7) [reply quotas/business_rules.py:253]: I wonder if there is scope in instituting our own business rules similar to how we have with the true TARIC spec ones? Even if we store this somewhere accessible - food for thought

**Pull request merged**